File: TOOLS/lua/observe-all.lua
-- Test script for property change notification mechanism.
-- Note that watching/reading some properties can be very expensive, or
-- require the player to synchronously wait on network (when playing
-- remote files), so you should in general only watch properties you
-- are interested in.

local utils = require("mp.utils")

function observe(name)
    mp.observe_property(name, "native", function(name, val)
        print("property '" .. name .. "' changed to '" ..
              utils.to_string(val) .. "'")
    end)
end

for i,name in ipairs(mp.get_property_native("property-list")) do
    observe(name)
end

for i,name in ipairs(mp.get_property_native("options")) do
    observe("options/" .. name)
end

File: TOOLS/python/observe-all.py
# Test script for property change notification mechanism.
# Note that watching/reading some properties can be very expensive, or
# require the player to synchronously wait on network (when playing
# remote files), so you should in general only watch properties you
# are interested in.

from mpvclient import mpv  #type: ignore


def observe(name):
    @mpv.observe_property(name, mpv.MPV_FORMAT_NODE)
    def func(val):
        mpv.info("property '" + name + "' changed to '" +
              str(val) + "'")


for name in mpv.get_property_node("property-list"):
    observe(name)
    # if name not in ["osd-sym-cc", "osd-ass-cc", "term-clip-cc",
    #     "screenshot-template"  # fails with: *** %n in writable segment detected ***
    # ]:
    #     observe(name)


for name in mpv.get_property_node("options"):
    observe(name)
    # if name not in ["screenshot-template"]:
    #     observe("options/" + name)
